test/unit/utils/jwt.spec.ts

```diff
@@ -709,24 +709,24 @@ describe('JwksFetcher', () => {
 
       return keyFetcher.fetchPublicKeys().then(() => {
         expect(https.request).to.have.been.calledOnce;
-        clock!.tick((ONE_DAY_IN_SECONDS - 1) * 1000);
+        clock!.tick((SIX_HOURS_IN_SECONDS - 1) * 1000);
         return keyFetcher.fetchPublicKeys();
       }).then(() => {
         expect(https.request).to.have.been.calledOnce;
-        clock!.tick(ONE_DAY_IN_SECONDS * 1000); // 24 hours in milliseconds
+        clock!.tick(SIX_HOURS_IN_SECONDS * 1000); // 6 hours in milliseconds
```

Comment:
I don't think the assertion on clock (`clock!`) is necessary here.

Reply:
We need it because of the way `clock` is defined in these tests: `let clock: sinon.SinonFakeTimers | undefined;`
We set it to `undefined` after each test case:
```
  afterEach(() => {
    if (clock) {
      clock.restore();
      clock = undefined;
    }
  });
```
I think just doing `clock.restore();` should be enough here. Let me know if you would like me to clean up all the test cases.

Reply:
Does the compiler actually throw for these instances? We initialize clock as follows in some these tests:

```
clock = sinon.useFakeTimers(1000);
```

After that compiler should see that `clock` is no longer `undefined`.

Reply:
Yeah, even after initializing the compiler throws if the instance is used inside a promise chain.